Feature/images

leonardoAnjos16 opened the pull request:
#### What is the purpose of this pull request?

To create an action for downloading images from MDX files and replacing their URL with a GitHub URL that leads to this repository

Documented [here](https://www.notion.so/vtexhandbook/Gerenciar-imagens-dentro-dos-MDXs-2d980e356f3245d5809ae03033ef4866)

#### What problem is this solving?

The tech writers will write MDX files using absolute and relative paths for images. But those files will be used in the Developers Portal application, with no access to those images. We need to update those paths to GitHub URLs so that the Developers Portal application knows where to find the images. Additionally, we need to download images from external sources to this repository and update the URLs in the MDX files accordingly.

#### How should this be manually tested?

The action should run on every pull request, looking for added/changed markdown or MDX files and downloading the images and updating their URLs inside the files. To test it, you may create a fork of this repository with these changes, open a pull request with new markdown or MDX files (preferably with image blocks inside) and check if the action runs and does what it is supposed to do.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

amandascm reviewed (commented):
Nice contribution 🎇 
I tried to do the fork test by adding a new raw markdown file (with no slug in its begining) but also applied some changes in one of the existing files (`add-to-cart-button-performance.md`). The workflow is running completely, but I got a `"Couldn't download some of the images, leaving file as it is..."` message for both images I've added (screenshot below).

Maybe you can help me understand what is going on or if I should test it another way ;)

[The opened PR in my fork (with main branch already including the new script and workflow)](https://github.com/amandascm/dev-portal-content/pull/3)

<img width="590" alt="image" src="https://user-images.githubusercontent.com/39717968/191351810-2ef0d156-3f9a-41ca-8519-bbd52f516571.png">

leonardoAnjos16 commented:
> Nice contribution 🎇 I tried to do the fork test by adding a new raw markdown file (with no slug in its begining) but also applied some changes in one of the existing files (`add-to-cart-button-performance.md`). The workflow is running completely, but I got a `"Couldn't download some of the images, leaving file as it is..."` message for both images I've added (screenshot below).
> 
> Maybe you can help me understand what is going on or if I should test it another way ;)
> 
> [The opened PR in my fork (with main branch already including the new script and workflow)](https://github.com/amandascm/dev-portal-content/pull/3)
> 
> <img alt="image" width="590" src="https://user-images.githubusercontent.com/39717968/191351810-2ef0d156-3f9a-41ca-8519-bbd52f516571.png">

Nice catches, @amandascm ! That message is logged for every MDX file with at least one image that couldn't be downloaded. I reproduced your steps on my fork repo and realized the images weren't being downloaded because the script got the file extension wrong (with slashes and some other special characters). I fixed it by asserting the output of the getExtension function is a valid file extension and returning the default 'png' otherwise. Also, the script wasn't replacing URLs inside img tags yet, so I added support for that now. Can you take another look, please?

amandascm reviewed (approved):
LGTM 🚀 
Everything is working fine!! 
([tested it in my fork again](https://github.com/amandascm/dev-portal-content/pull/4))

RobsonOlv reviewed (approved):
LGTM :rocket:
